Decred Wallet/Features/Navigation Menu/CustomTabMenuView.swift

```diff
@@ -0,0 +1,125 @@
+        super.init(coder: aDecoder)
+    }
+    
+    convenience init(menuItems: [MenuItem], frame: CGRect) {
+        self.init(frame: frame)
+        self.axis = .horizontal
+        self.layer.backgroundColor = UIColor.white.cgColor
+        
+        for menuItem in menuItems {
+            let itemView = self.createTabItemButton(for: menuItem)
+            itemView.translatesAutoresizingMaskIntoConstraints = false
+            itemView.clipsToBounds = true
+            self.addArrangedSubview(itemView)
+        }
+        
+        self.setNeedsLayout()
+        self.layoutIfNeeded()
+        self.activateTab(viewId: self.activeTabIndex)
+    }
+    
+    // Create a custom nav menu item
+    func createTabItemButton(for menuItem: MenuItem) -> UIView {
+        let tabBarItem = UIView(frame: CGRect.zero)
+        
+        let itemTitleLabel = UILabel(frame: CGRect.zero)
+        let itemIconView = UIImageView(frame: CGRect.zero)
+        
+        itemTitleLabel.font = UIFont(name: "Source Sans Pro", size: 13)
+        itemTitleLabel.text = menuItem.displayTitle
+        itemTitleLabel.numberOfLines = 0
+        itemTitleLabel.adjustsFontSizeToFitWidth = true
+        itemTitleLabel.textColor = UIColor.appColors.darkGray
+        itemTitleLabel.textAlignment = .center
+        itemTitleLabel.translatesAutoresizingMaskIntoConstraints = false
+        itemTitleLabel.clipsToBounds = true
+        
+        itemIconView.image = menuItem.icon!.withRenderingMode(.automatic)
+        itemIconView.translatesAutoresizingMaskIntoConstraints = false
+        itemIconView.clipsToBounds = true
+        
+        tabBarItem.layer.backgroundColor = UIColor.white.cgColor
+        tabBarItem.addSubview(itemIconView)
+        tabBarItem.addSubview(itemTitleLabel)
+        tabBarItem.translatesAutoresizingMaskIntoConstraints = false
+        tabBarItem.clipsToBounds = true
+        
+        let constraints = [
+            itemIconView.heightAnchor.constraint(equalToConstant: 25), // Fixed height for our tab item icon
+            itemIconView.widthAnchor.constraint(equalToConstant: 25), // Fixed width for our tab item icon
+            itemIconView.centerXAnchor.constraint(equalTo: tabBarItem.centerXAnchor),
+            itemIconView.topAnchor.constraint(equalTo: tabBarItem.topAnchor, constant: 8), // Position menu item icon 8pts from the top of it's parent view
+            itemIconView.leadingAnchor.constraint(equalTo: tabBarItem.leadingAnchor, constant: 35),
+            
+            itemTitleLabel.heightAnchor.constraint(equalToConstant: 13), // Fixed height for title label
+            itemTitleLabel.widthAnchor.constraint(equalTo: tabBarItem.widthAnchor), // Position label full width across tab bar item
+            itemTitleLabel.topAnchor.constraint(equalTo: itemIconView.bottomAnchor, constant: 4), // Position title label 4pts below item icon
+        ]
+        NSLayoutConstraint.activate(constraints)
```

Comment:
These constraints are producing constraint satisfaction issues on the log output. Something isn't right and needs to be fixed.

I want to recommend designing this custom view and related custom views in a way that it can be previewed from interface builder. I'll run an experiment on this and let you know how it goes.

Reply:
All constraint satisfaction issues are resolved in 2766f8f